lib/logstash/bundler.rb

```diff
@@ -23,6 +23,28 @@ def reset!
 
 module LogStash
   module Bundler
+    # Take a gem package and extract it to a specific target
+    # @param [String] Gem file, this must be a path
+    # @param [String, String] Return a Gem::Package and the installed path
+    def self.unpack(file, path)
+      require "rubygems/package"
+      require "securerandom"
+
+      # We are creating a random directory per extract,
+      # if we dont do this bundler will not trigger download of the dependencies.
+      # Use case is:
+      # - User build his own gem with a fix
+      # - User doesnt increment the version
+      # - User install the same version but different code or dependencies multiple times..
+      basename  = ::File.basename(file, '.gem') 
+      unique = SecureRandom.hex(4)
+      target_path = ::File.expand_path(::File.join(path, unique, basename))
```

Comment:
I can't use `stud` here, it force the use of the `/tmp` directory of the OS.
The extracted files need to exist after restart so It is not really a temporary directory.